cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/aws/aws-mc-common/management-credential-step/ManagementCredentials.tsx

```diff
@@ -111,8 +112,33 @@ function ManagementCredentials(props: Partial<StepProps>) {
     }, [connectionStatus]);
 
     useEffect(() => {
-        AwsOrchestrator.initNodeProfile({ awsState, awsDispatch, errorObject, setErrorObject });
-    }, []);
+        if (awsState[STORE_SECTION_FORM][AWS_FIELDS.NODE_PROFILE]) {
+            AwsOrchestrator.initControlPlaneNodeType(
+                { awsState, awsDispatch, errorObject, setErrorObject },
+                awsState[STORE_SECTION_FORM][AWS_FIELDS.NODE_PROFILE]
+            );
+        }
+    }, [awsState[STORE_SECTION_FORM][AWS_FIELDS.NODE_PROFILE]]);
+
+    useEffect(() => {
+        if (connectionStatus === CONNECTION_STATUS.CONNECTED) {
+            AwsOrchestrator.initAvailabilityZones({ awsState, awsDispatch, errorObject, setErrorObject });
+        }
+    }, [connectionStatus]);
+
+    useEffect(() => {
+        if (awsState[STORE_SECTION_RESOURCES][AWS_FIELDS.AVAILABILITY_ZONES] !== undefined) {
```

Comment:
nit: I think you can remove `!== undefined` from the conditional

Reply:
removed  `!== undefined`